storage/remote/queue_manager.go

```diff
@@ -343,8 +343,12 @@ func (t *QueueManager) StoreSeries(series []tsdb.RefSeries, index int) {
 	t.seriesMtx.Lock()
 	defer t.seriesMtx.Unlock()
 	for ref, labels := range temp {
-		t.seriesLabels[ref] = labels
 		t.seriesSegmentIndexes[ref] = index
+
+		if orig, ok := t.seriesLabels[ref]; ok {
```

Comment:
This takes a bit to understand. A comment would help.

Reply:
Done.